src/components/ResultContainer.tsx

```diff
@@ -0,0 +1,220 @@
+import { useState, useEffect } from 'react';
+import { Button, FormControlLabel, Checkbox, Typography } from '@mui/material';
+import ResultCard from './ResultCard';
+import { Result } from '../utils/types';
+import DownloadResultButton from './DownloadResultButton';
+import NBDialog from './NBDialog';
+
+function ResultContainer({ result }: { result: Result[] | null }) {
+  const [download, setDownload] = useState<string[]>([]);
+  const [openModal, setOpenModal] = useState(false);
+
+  // TODO: deal with erros
+  function updateDownload(checked: boolean, id: string) {
+    setDownload((currDownload) => {
+      const newDownload = checked
+        ? [...currDownload, id]
+        : currDownload.filter((downloadID) => downloadID !== id);
+      return newDownload;
+    });
+  }
+
+  function selectAll() {
+    if (result) {
+      return (
+        result.length === download.length && result.every((r) => download.includes(r.dataset_uuid))
+      );
+    }
+    return false;
+  }
```

Comment:
I really think this should be a regular variable  / computed property. Right now there is no single variable that tells you what the status of `checked` is, so whenever I want to know, I need to call this function. 

And all functions that rely on knowing what the current value of `checked` is, need to be given that value where they are invoked. I think it'd be much cleaner if we had `const checked = ...` up top and then just use that wherever `checked` needs to be accessed

Reply:
I think its form (variable or function) doesn't change whether it is a computed property or not.
I do, however, see your point and the benefit of changing it to a constant to avoid the overhead caused by calling the function.